apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/repository/model/GenericResponse.kt

```diff
@@ -0,0 +1,42 @@
+package com.twilio.livevideo.app.repository.model
+
+import com.google.gson.annotations.SerializedName
+
+data class GenericResponse(
+
+    @SerializedName("verified")
+    val isVerified: Boolean? = null,
+
+    @SerializedName("success")
+    val isSuccess: Boolean? = null,
+
+    @SerializedName("sent")
+    val isSent: Boolean? = null,
+
+    @SerializedName("removed")
+    val isRemoved: Boolean? = null,
+
+    @SerializedName("token")
+    val token: String? = null
```

Comment:
I think it's great to have a reusable `ErrorResponse` and maybe even a way to reuse the token response since it is the same for 3 endpoints.

But I wonder about having a single response that works for any endpoint in the app like this. It seems like it is less type safe because everything has to be optional. And while this approach may work in this reference app because the responses are simple and there are not that many requests, I wonder if this is standard best practice.

What are the benefits of this approach? Is this a popular Android convention? Or am I misunderstanding something?

Reply:
Yeah I understand. I think it's probably best to map each response, so it's more conventional for others looking at the code.

Reply:
Now that the responses are separated I think a lot of the fields don't need to be optional anymore. For example:
```
val isVerified: Boolean? = null
```